tests/queries/0_stateless/02160_special_functions.sql

```diff
@@ -41,16 +41,4 @@ SELECT TIMESTAMPSUB(DATE '2022-01-01', INTERVAL 1 YEAR);
 SELECT DATE_DIFF(YEAR, DATE '2021-01-01', DATE '2022-01-01');
 SELECT DATEDIFF(YEAR, DATE '2021-01-01', DATE '2022-01-01');
 
-SELECT DATEDIFF(millisecond, '2021-01-01'::Date, '2021-01-02'::Date);
-SELECT DATEDIFF(millisecond, '2021-01-01'::Date, '2021-01-03'::Date32);
-SELECT DATEDIFF(millisecond, '2021-01-01'::Date, '2021-01-02 00:01:01'::DateTime);
-SELECT DATEDIFF(millisecond, '2021-01-01'::Date, '2021-01-02 00:00:01.299'::DateTime64);
-SELECT DATEDIFF(millisecond, '2021-01-01 23:59:59.299'::DateTime64, '2021-01-02'::Date);
-SELECT DATEDIFF(millisecond, '2021-01-01 23:59:59.299999'::DateTime64(6), '2021-01-02'::Date);
-SELECT DATEDIFF(millisecond, '2021-01-01 23:59:59.2'::DateTime64(1), '2021-01-02'::Date);
-SELECT DATEDIFF(microsecond, '2021-01-01 23:59:59.899999'::DateTime64(6), '2021-01-02 00:01:00.100200300'::DateTime64(9));
-
-SELECT DATEDIFF(microsecond, '1969-12-31 23:59:59.999950'::DateTime64(6), '1970-01-01 00:00:00.000010'::DateTime64(6));
```

Comment:
Maybe it worth to simply pin to UTC here? 

```sql
SELECT dateDiff('microsecond', CAST('1969-12-31 23:59:59.999950', 'DateTime64(6, \'America/Mazatlan\')'), CAST('1970-01-01 00:00:00.000010', 'DateTime64(6, \'America/Mazatlan\')'))

┌─dateDiff('microsecond', CAST('1969-12-31 23:59:59.999950', 'DateTime64(6, \'America/Mazatlan\')'), CAST('1970-01-01 00:00:00.000010', 'DateTime64(6, \'America/Mazatlan\')'))─┐
│                                                                                                                                                                   -3599999940 │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

SELECT dateDiff('microsecond', CAST('1969-12-31 23:59:59.999950', 'DateTime64(6, \'UTC\')'), CAST('1970-01-01 00:00:00.000010', 'DateTime64(6, \'UTC\')'))

┌─dateDiff('microsecond', CAST('1969-12-31 23:59:59.999950', 'DateTime64(6, \'UTC\')'), CAST('1970-01-01 00:00:00.000010', 'DateTime64(6, \'UTC\')'))─┐
│                                                                                                                                                  60 │
└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Since the underlying problem is not in this PR https://github.com/ClickHouse/ClickHouse/issues/42653

Cc: @rvasin

Reply:
1. @azat - yes, all usages of DateTime64 tests related to this feature must in with timezone. And we usually use UTC timezone in all other tests.
2. #42653 exactly I did some deep analysis for **all** time zones for different date time ranges. It's better to avoid in tests about Unix epoch. Because Unix epoch is the time when transitions took place for some time zones. And there are could be some time ranges which in fact did not exist in time for certain time zones (because of time transitions). I analyzed how different DBMSs handle such time ranges differently. 
3. @kardymonds  will try to correct the flaky test on Monday.

Reply:
BTW maybe tricky timezones should be used in the flaky checks, so that such PRs will not be merged in advance

Reply:
@azat Hi, can you, please, tell me how to find (by himself) output flaky tests.

Reply:
If the question is "Where in github actions flaky checks?" then everything that has "flaky" in it's names, i.e. for your original PR:

>[Stateless tests flaky check (asan) — fail: 0, passed: 303](https://s3.amazonaws.com/clickhouse-test-reports/51291/3237169c9a320bc4bc7d5eae3925755a4d23859f/stateless_tests_flaky_check__asan_.html)

And the output is in [`run.log`](https://s3.amazonaws.com/clickhouse-test-reports/51291/3237169c9a320bc4bc7d5eae3925755a4d23859f/stateless_tests_flaky_check__asan_/run.log)

_(you don't need integration flaky check)_

But if the question is "Where it is implemented in code?" then here - https://github.com/ClickHouse/ClickHouse/blob/44929b7d2873704c4df4bc54ddfbd4c36ef322de/tests/ci/functional_test_check.py#L104

Reply:
Ok, But my pull request didn't (seems) have a flaky in CI. They appeared in the master (or in foreign branches). How to find out by the name of the test that it was (now is reverted) a flaky and its output (diff / lines).